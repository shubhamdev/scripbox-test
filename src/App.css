.App {
  text-align: center;
}
 main{
  min-height: 90vh
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.header{
  display: flex;
  height: 8vh;
  width: 100vw;
  background-color: royalblue;
}

.footer{
  margin-top: 5px;
  padding: 5px;
  background-color: rgb(235, 195, 64);
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
}