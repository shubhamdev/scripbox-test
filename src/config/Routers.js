import React from "react";
import { Link, Switch } from "react-router-dom";
import PrivateRoute from "./PrivateRoutes";
import PublicRoute from "./PublicRoutes";
import { Layout } from "antd";
const { Header, Content } = Layout;
const LoginPage = React.lazy(() => import("../components/loginPage/LoginPage"));
const ChallengesPage = React.lazy(() =>
  import("../components/challengesPage/ChallengesPage")
);
const DashboardPage = React.lazy(() =>
  import("../components/dashboardPage/DashboardPage")
);

const HomePage = React.lazy(() => import("../components/homePage/HomePage"));
function NoMatch() {
  return (
    <div
      style={{
        alignItems: "center",
        display: "flex",
        justifyContent: "center",
        padding: "1rem",
        height: "90vh",
      }}
    >
      <h3>
        No match for this route
        <Link to="/"> Login</Link>
      </h3>
    </div>
  );
}

function Routers() {
  return (
    <Layout>
      <Header>
        <HomePage />
      </Header>
      <Content style={{ height: "100bh" }}>
        <Switch>
          <PublicRoute
            restricted={false}
            component={LoginPage}
            path="/"
            exact
          />
          <PrivateRoute component={ChallengesPage} path="/challenge" exact />
          <PrivateRoute component={DashboardPage} path="/dashboard" exact />
          <PublicRoute path="*">
            <NoMatch />
          </PublicRoute>
        </Switch>
      </Content>
    </Layout>
  );
}

export default Routers;
