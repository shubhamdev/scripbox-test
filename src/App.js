import React, { Suspense, createContext } from "react";
import "./App.css";
import PageContextProvider from "./PageContextProvider";
const Routers = React.lazy(() => import("./config/Routers"));
export const PageContext = createContext();

function App() {
  return (
    <Suspense fallback="Loading">
      <PageContextProvider>
        <Routers />
      </PageContextProvider>
    </Suspense>
  );
}

export default App;
