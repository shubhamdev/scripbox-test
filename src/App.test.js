import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { shallow } from "enzyme";
import App from "./App";
import Routers from "./config/Routers";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Render App component", () => {
  it('should render App component correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });
  it("should render App component correctly with no props", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it('should render Routers component correctly in "debug" mode', () => {
    const component = shallow(<Routers debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render Routers component correctly with no props", () => {
    const component = shallow(<Routers />);
    expect(component).toMatchSnapshot();
  });
});
