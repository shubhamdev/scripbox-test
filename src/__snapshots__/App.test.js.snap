// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App component should render App component correctly in "debug" mode 1`] = `
<Suspense
  debug={true}
  fallback="Loading"
>
  <PageContextProvider>
    <lazy />
  </PageContextProvider>
</Suspense>
`;

exports[`Render App component should render App component correctly with no props 1`] = `
<Suspense
  fallback="Loading"
>
  <PageContextProvider>
    <lazy />
  </PageContextProvider>
</Suspense>
`;

exports[`Render App component should render Routers component correctly in "debug" mode 1`] = `
<Layout>
  <Header>
    <lazy />
  </Header>
  <Content
    style={
      Object {
        "height": "100bh",
      }
    }
  >
    <Switch>
      <PublicRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/"
        restricted={false}
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/challenge"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/dashboard"
      />
      <PublicRoute
        path="*"
      >
        <NoMatch />
      </PublicRoute>
    </Switch>
  </Content>
</Layout>
`;

exports[`Render App component should render Routers component correctly with no props 1`] = `
<Layout>
  <Header>
    <lazy />
  </Header>
  <Content
    style={
      Object {
        "height": "100bh",
      }
    }
  >
    <Switch>
      <PublicRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/"
        restricted={false}
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/challenge"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/dashboard"
      />
      <PublicRoute
        path="*"
      >
        <NoMatch />
      </PublicRoute>
    </Switch>
  </Content>
</Layout>
`;
